File: src/test/java/com/automation/framework/pages/BasePage.java
package com.automation.framework.pages;

import com.automation.framework.driver.DriverManager;
import com.automation.framework.utils.ConfigReader;
import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;

public abstract class BasePage {

  protected final WebDriverWait wait;

  protected BasePage() {
    int seconds = ConfigReader.getInt("explicitWaitSeconds", 10);
    this.wait = new WebDriverWait(DriverManager.getDriver(), Duration.ofSeconds(seconds));
  }

  protected WebElement visible(By by) {
    return wait.until(ExpectedConditions.visibilityOfElementLocated(by));
  }

  protected void type(By by, String value) {
    WebElement el = visible(by);
    el.clear();
    el.sendKeys(value);
  }

  protected void click(By by) {
    wait.until(ExpectedConditions.elementToBeClickable(by)).click();
  }

  protected String text(By by) {
    return visible(by).getText();
  }
}

File: python/framework/pages/base_page.py
from selenium.webdriver.support import expected_conditions as ec
from selenium.webdriver.support.ui import WebDriverWait

from framework.driver.driver_manager import DriverManager
from framework.utils.config_reader import get_int


class BasePage:
    def __init__(self):
        seconds = get_int("explicitWaitSeconds", 10)
        self.wait = WebDriverWait(DriverManager.get_driver(), seconds)

    def visible(self, locator):
        return self.wait.until(ec.visibility_of_element_located(locator))

    def type(self, locator, value):
        element = self.visible(locator)
        element.clear()
        element.send_keys(value)

    def click(self, locator):
        self.wait.until(ec.element_to_be_clickable(locator)).click()

    def text(self, locator):
        return self.visible(locator).text
